raiden/network/proxies/token_network.py

```diff
@@ -708,51 +657,89 @@ def set_total_deposit(
         """
         typecheck(total_deposit, int)
 
-        with self.channel_operations_lock[partner], self.deposit_lock:
-            previous_total_deposit = self._detail_participant(
-                channel_identifier=channel_identifier,
-                detail_for=self.node_address,
-                partner=partner,
-                block_identifier=given_block_identifier,
-            ).deposit
+        token_address = self.token_address()
+        token = Token(
```

Comment:
Yeah, also, we have to hold the `token` lock to perform this operation, otherwise it is possible to race with calls to `approve` and `transfer` and something will fail. (Please, when fixing double check the acquisition order avoid deadlocks)